models/issue_assignees.go

```diff
@@ -0,0 +1,232 @@
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// MakeAssigneeList concats a string with all names of the assignees. Useful for logs.
+func MakeAssigneeList(issue Issue) (AssigneeList string, err error) {
+	err = issue.loadAssignees(x)
+	if err != nil {
+		return "", err
+	}
+
+	for in, assignee := range issue.Assignees {
+		AssigneeList += assignee.Name
+
+		if len(issue.Assignees) > (in + 1) {
+			AssigneeList += ", "
+		}
+	}
+	return
+}
+
+// ClearAssigneeByUserID deletes all assignments of an user
+func clearAssigneeByUserID(sess *xorm.Session, userID int64) (err error) {
+	_, err = sess.Delete(&IssueAssignees{AssigneeID: userID})
+	return
+}
+
+// UpdateAssignee deletes or adds an assignee to an issue
+func UpdateAssignee(issue *Issue, doer *User, assigneeID int64) (err error) {
+	return issue.ChangeAssignee(doer, assigneeID)
+}
+
+// ChangeAssignee changes the Assignee of this issue.
+func (issue *Issue) ChangeAssignee(doer *User, assigneeID int64) (err error) {
+	sess := x.NewSession()
+	defer sess.Close()
+
+	return issue.changeAssignee(sess, doer, assigneeID)
+}
+
+func (issue *Issue) changeAssignee(sess *xorm.Session, doer *User, assigneeID int64) (err error) {
+
+	// Update the assignee
+	removed, err := updateIssueAssignee(sess, issue, assigneeID)
+	if err != nil {
+		return fmt.Errorf("UpdateIssueUserByAssignee: %v", err)
+	}
+
+	// Repo infos
+	if err = issue.loadRepo(sess); err != nil {
+		return fmt.Errorf("loadRepo: %v", err)
+	}
+
+	// Comment
+	if _, err = createAssigneeComment(sess, doer, issue.Repo, issue, assigneeID, removed); err != nil {
+		return fmt.Errorf("createAssigneeComment: %v", err)
+	}
+
+	if issue.IsPull {
+		issue.PullRequest = &PullRequest{Issue: issue}
+		apiPullRequest := &api.PullRequestPayload{
+			Index:       issue.Index,
+			PullRequest: issue.PullRequest.APIFormat(),
+			Repository:  issue.Repo.APIFormat(AccessModeNone),
+			Sender:      doer.APIFormat(),
+		}
+		if removed {
+			apiPullRequest.Action = api.HookIssueUnassigned
+		} else {
+			apiPullRequest.Action = api.HookIssueAssigned
+		}
+		if err := PrepareWebhooks(issue.Repo, HookEventPullRequest, apiPullRequest); err != nil {
+			log.Error(4, "PrepareWebhooks [is_pull: %v, remove_assignee: %v]: %v", issue.IsPull, removed, err)
+			return nil
+		}
+	}
+	go HookQueue.Add(issue.RepoID)
+	return nil
+}
+
+// UpdateAPIAssignee is a helper function to add or delete one or multiple issue assignee(s)
+// Deleting is done the Github way (quote from their api documentation):
+// https://developer.github.com/v3/issues/#edit-an-issue
+// "assignees" (array): Logins for Users to assign to this issue.
+// Pass one or more user logins to replace the set of assignees on this Issue.
+// Send an empty array ([]) to clear all assignees from the Issue.
+func UpdateAPIAssignee(issue *Issue, oneAssignee string, multipleAssignees []string, doer *User) (err error) {
+	// Clear everyone
+	err = ClearAssigneesByIssue(issue)
```

Comment:
why clear everyone?

Reply:
Because that way we it'll remove everyone which we didn't pass withing our array. 
The behavior of the API I intended is to remove everyone you don't pass to it.
This way is the easiest solution, another one would have been to loop through the array, look who of the (new) assignees is already assigned, create a different array with everyone who isn't in that list anymore, loop through that and remove everyone.

Another way to remove or add assignees would have been a seperate api endpoint for that, but I didn't wanted to add too much API in this PR.

Reply:
You can `delete where not in` to do that.